File: frontend/src/types/call.ts
// Call record types

export interface CallRecord {
  id: string;
  conversation_id: string;
  direction: "inbound" | "outbound";
  channel: string;
  status: "initiated" | "ringing" | "in_progress" | "completed" | "failed" | "busy" | "no_answer";
  from_number?: string;
  to_number?: string;
  contact_name?: string;
  contact_id?: number;
  duration_seconds?: number;
  recording_url?: string;
  transcript?: string;
  agent_id?: string;
  agent_name?: string;
  is_ai?: boolean;
  booking_outcome?: string;
  created_at: string;
  // Optional fields for active calls
  started_at?: string;
  answered_at?: string;
  ended_at?: string;
}

File: backend/app/schemas/call.py
"""Call schemas for voice call API endpoints."""

import uuid
from datetime import datetime

from pydantic import BaseModel


class CallCreate(BaseModel):
    """Request to initiate a call."""

    to_number: str
    from_phone_number: str
    contact_phone: str | None = None
    agent_id: uuid.UUID | None = None


class CallResponse(BaseModel):
    """Voice call response."""

    id: uuid.UUID
    conversation_id: uuid.UUID
    direction: str  # inbound/outbound
    channel: str
    status: str  # queued/ringing/answered/completed/failed
    duration_seconds: int | None
    recording_url: str | None
    transcript: str | None  # JSON array of transcript entries
    created_at: datetime
    # Phone numbers from conversation
    from_number: str | None = None
    to_number: str | None = None
    # Contact info
    contact_name: str | None = None
    contact_id: int | None = None
    # Agent info
    agent_id: uuid.UUID | None = None
    agent_name: str | None = None
    is_ai: bool = False
    booking_outcome: str | None = None


class PaginatedCalls(BaseModel):
    """Paginated calls response."""

    items: list[CallResponse]
    total: int
    page: int
    page_size: int
    pages: int
    # Aggregate stats across all matching calls (not just current page)
    completed_count: int = 0
    total_duration_seconds: int = 0
